core/src/main/scala/pureconfig/BasicReaders.scala

```diff
@@ -114,6 +116,35 @@ trait DurationReaders {
 }
 
 /**
+ * Trait containing `ConfigReader` instances for Java and Scala arbitrary-precision numeric types.
+ */
+trait NumericReaders {
+
+  // BigDecimal String parsing copied from Scala 2.11.
+  // The Scala 2.10 bigDecimal class is hard-coded to use 128-bits of prevision when parsing BigDecimal from String.
+  // This causes truncation for large values.
+  // Scala 2.11+ fixed this bug by adapting precision to accommodate all digits in the string.
+  private def exact(repr: JavaBigDecimal): BigDecimal = {
```

Comment:
Yeah, I shouldn't have pushed that late at night. I wonder why if even compiled. Anyway, it's fixed.